packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -134,6 +134,17 @@
       visibility: 'hidden !important'
     };
 
+    /**
+     * Initializes images state and resets when the result changes.
+     *
+     * @internal
+     */
+    @Watch('result', { immediate: true })
```

Comment:
I would probably watch `result.images`, just in case the images arrays mutates.

Reply:
That should check both `result` and `images`.

Reply:
Do you suggest to change watch for `'result.images'`?

Reply:
Yes: `@Watch('result.images', { immediate: true })`